client/src/lib/data.ts

```diff
@@ -0,0 +1,70 @@
+export type Movie = {
+  title: string;
+  summary: string;
+  link: string;
+  rating: number;
+  movieId?: number;
+};
+
+export async function getMovies(): Promise<Movie[] | null> {
+  const url = "http://localhost:3000/api/movies";

```

Comment:
Don't ever hard-code the domain name and port. With Vite, you just use `/api/movies` and then you configure vite.config.ts with a server.proxy field that forwards the api requests to your backend.